### PR TITLE
Some type hints

### PR DESCRIPTION
**Pull request by elacuesta**
Just a few annotations to remove exceptions from `setup.cfg`

### CODE REVIEW

File: scrapy/http/cookies.py [hunk 1]
```diff
@@ -1,10 +1,14 @@
+import re
 import time
-from http.cookiejar import CookieJar as _CookieJar, DefaultCookiePolicy, IPV4_RE
+from http.cookiejar import CookieJar as _CookieJar, DefaultCookiePolicy
 
 from scrapy.utils.httpobj import urlparse_cached
 from scrapy.utils.python import to_unicode
 
 
+IPV4_RE = re.compile(r"\.\d+$", re.ASCII)
```
Comment (elacuesta) [inline scrapy/http/cookies.py:9]: Taken from https://github.com/python/cpython/blob/3.9/Lib/http/cookiejar.py#L527

`IPV4_RE` is neither in [`__all__`](https://github.com/python/cpython/blob/3.9/Lib/http/cookiejar.py#L28-L29) nor in the [docs](https://docs.python.org/3/library/http.cookiejar.html).

Comment (kmike) [reply scrapy/http/cookies.py:9]: Would you mind adding something simiilar as a code comment?

File: scrapy/__init__.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 
 # Scrapy and Twisted versions
-__version__ = pkgutil.get_data(__package__, 'VERSION').decode('ascii').strip()
+__version__ = (pkgutil.get_data(__package__, "VERSION") or b"").decode("ascii").strip()
```
Comment (Gallaecio) [inline scrapy/__init__.py:25]: :lipstick: Why the change of quotes? I see the line below also uses single quotes.

**Review by Gallaecio — APPROVED** [review]

File: scrapy/__init__.py (hunk 2 above)
Comment (elacuesta) [reply scrapy/__init__.py:25]: It's a subtle step towards #4658 :wink:

File: scrapy/utils/httpobj.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
 """Helper functions for scrapy.http objects (Request, Response)"""
 
-import weakref
 from urllib.parse import urlparse
+from weakref import WeakKeyDictionary
 
 
-_urlparse_cache = weakref.WeakKeyDictionary()
+_urlparse_cache: WeakKeyDictionary = WeakKeyDictionary()
```
Comment (kmike) [inline scrapy/utils/httpobj.py:7]: Would it be possible to type annotate keys/values here, and other similar places? It is a chance to make type annotations a bit more useful for reading (and probably using) the code; otherwise I suspect we'll never get to this - mypy error is silenced, all is good :)

Comment (elacuesta) [reply scrapy/utils/httpobj.py:7]: I looked into [`typing`](https://docs.python.org/3/library/typing.html) and couldn't find it, so I decided to only annotate the "outer" type. 
According to [this comment](https://bugs.python.org/issue38756#msg356318), it should be possible to annotate it as a string (`from __future__ import annotations` doesn't work in py36).
I'll update this shortly.

File: scrapy/commands/__init__.py [hunk 4]
```diff
@@ -15,7 +17,7 @@ class ScrapyCommand:
     crawler_process = None
 
     # default settings to be used for this command instead of global defaults
-    default_settings = {}
+    default_settings: Dict[str, Any] = {}
```
Comment (kmike) [inline scrapy/commands/__init__.py:20]: Should keys be str? We support callables as keys in settings now.

File: scrapy/utils/response.py [hunk 5]
```diff
@@ -60,7 +67,7 @@ def response_httprepr(response):
     return b"".join(values)
 
 
-def open_in_browser(response, _openfunc=webbrowser.open):
+def open_in_browser(response: Response, _openfunc: Callable[[str], Any] = webbrowser.open) -> Any:
```
Comment (kmike) [inline scrapy/utils/response.py:70]: Nitpick: return type is not Any, but whatever _openfunc returns. TypeVar would be more precise here.

File: scrapy/commands/__init__.py (hunk 4 above)
Comment (elacuesta) [reply scrapy/commands/__init__.py:20]: Do you mean after #3873? Those are values, keys can be other objects but only as sub-keys within other settings (`ITEM_PIPELINES`, `SPIDER_MIDDLEWARES`, etc)

File: scrapy/utils/response.py (hunk 5 above)
Comment (elacuesta) [reply scrapy/utils/response.py:70]: Makes sense, but I think I need a bit of assistance. If I do:
```python
OpenInBrowserTypeVar = TypeVar("OpenInBrowserTypeVar")

def open_in_browser(
    response: Union[HtmlResponse, TextResponse],
    _openfunc: Callable[[str], OpenInBrowserTypeVar] = webbrowser.open,
) -> OpenInBrowserTypeVar:
```
I get
```
error: Incompatible default for argument "_openfunc" (default has type "Callable[[str, int, bool], bool]", argument has type "Callable[[str], OpenInBrowserTypeVar]")
```
because [`webbrowser.open`](https://docs.python.org/3/library/webbrowser.html) takes `url, new=0, autoraise=True`. However, the function is called with `_openfunc(f"file://{fname}")`, passing only a `str`. Of course I could do
```
_openfunc: Callable[[str], OpenInBrowserTypeVar] = webbrowser.open,  # type: ignore
```
but I'd like to avoid that.

I haven't found a solution so far, my next step would be to ask in SO, but maybe you know a way to solve it.

**Comment by kmike** [comment]
Thanks @elacuesta!
